### PR TITLE
changed the logic of the appearance of the global spinner

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**

### CODE REVIEW

File: src/app/components/spinner/spinner.interceptor.ts [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+
 export class SpinnerInterceptor implements HttpInterceptor {
     constructor(private spinnerService: SpinnerService) {}
     timer: NodeJS.Timer;
     intercept(req: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
-        if (
-            req.url.includes("/assets") ||
-            (req.body as { operationName: string }).operationName === "getDatasetListFlows"
-        ) {
+        const skipGlodalLoader = req.headers.get("skip-loading");
```
Comment (zaychenko-sergei) [inline src/app/components/spinner/spinner.interceptor.ts:13]: We have a place to define constants

File: src/app/components/spinner/spinner.interceptor.ts [hunk 2]
```diff
@@ -1,17 +1,17 @@
 import { SpinnerService } from "./spinner.service";
-import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from "@angular/common/http";
+import { HttpContextToken, HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from "@angular/common/http";
 import { Observable } from "rxjs";
 import { finalize } from "rxjs/operators";
 import { environment } from "../../../environments/environment";
 
+export const IS_PUBLIC_API = new HttpContextToken<boolean>(() => false);
```
Comment (zaychenko-sergei) [inline src/app/components/spinner/spinner.interceptor.ts:7]: Not used anywhere

File: src/app/components/spinner/spinner.interceptor.ts [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
 export class SpinnerInterceptor implements HttpInterceptor {
     constructor(private spinnerService: SpinnerService) {}
     timer: NodeJS.Timer;
     intercept(req: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
-        if (
-            req.url.includes("/assets") ||
-            (req.body as { operationName: string }).operationName === "getDatasetListFlows"
-        ) {
+        const skipGlodalLoader = req.headers.get("skip-loading");
+        if (req.url.includes("/assets") || (skipGlodalLoader && JSON.parse(skipGlodalLoader))) {
```
Comment (zaychenko-sergei) [inline src/app/components/spinner/spinner.interceptor.ts:14]: Avoid implicit conversion to boolean

File: src/app/app.module.ts [hunk 4]
```diff
@@ -104,13 +106,22 @@ const Services = [
                         headers: new HttpHeaders().set("Authorization", `Bearer ${accessToken}`),
                     });
                 }
+                return forward(operation);
+            });
 
+            const globalLoaderMiddleware: ApolloLink = new ApolloLink((operation: Operation, forward: NextLink) => {
+                const context = operation.getContext();
+                const skipLoading = Boolean(context.skipLoading);
+                const headers = context.headers as HttpHeaders;
```
Comment (zaychenko-sergei) [inline src/app/app.module.ts:115]: Might be undefined, you need to provide a default array

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]
There was also a request to disable spinner for auto-complete queries:
https://github.com/kamu-data/kamu-web-ui/issues/267

File: src/app/components/spinner/spinner.interceptor.ts (hunk 1 above)
Comment (dmitriy-borzenko) [reply src/app/components/spinner/spinner.interceptor.ts:13]: Fixed

File: src/app/components/spinner/spinner.interceptor.ts (hunk 2 above)
Comment (dmitriy-borzenko) [reply src/app/components/spinner/spinner.interceptor.ts:7]: Removed

File: src/app/components/spinner/spinner.interceptor.ts (hunk 3 above)
Comment (dmitriy-borzenko) [reply src/app/components/spinner/spinner.interceptor.ts:14]: Fixed

File: src/app/app.module.ts (hunk 4 above)
Comment (dmitriy-borzenko) [reply src/app/app.module.ts:115]: Fixed

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**